biz.aQute.bndlib/src/aQute/bnd/build/ProjectBuilder.java

```diff
@@ -594,29 +594,32 @@ private boolean isMaster(InfoRepository repo, String bsn, Version v) throws Exce
 	}
 
 	private RepositoryPlugin getReleaseRepo() {
-		String repoName = getProperty(Constants.RELEASEREPO);
 
-		List<RepositoryPlugin> repos = getPlugins(RepositoryPlugin.class);
-		for (RepositoryPlugin r : repos) {
-			if (r.canWrite()) {
-				if (repoName == null || r.getName()
-					.equals(repoName)) {
-					return r;
-				}
-			}
-		}
-		if (repoName == null)
+		String repoNames = getProperty(Constants.RELEASEREPO);
+
+		if (repoNames == null) {
 			error("Could not find a writable repo for the release repo (-releaserepo is not set)");
```

Comment:
The error message seems a bit wrong since it mentions writeable repos when no one has even looked for a writeable repo. I also thought we decided not to change behavior and just warn if -releaserepo was not set but still release to the first writeable repo.